### PR TITLE
Add support for owner filed in package manifest

### PR DESCRIPTION
**Pull request by ruflin**
In some cases it is important to know who the owner of a package is so this owner can be pinged for questions. In https://github.com/elastic/package-registry/issues/451 the different options around owner, authors, contributors were discussed. My take away from this discussions which lead to this PR:

* There are multiple authors over time
* Authors and contributors are mostly exchangeable terms
* There can only be 1 owner, even if it is a team.

This PR currently only implements `owner.github` because it is all we need. Later on we can add an array of `contributors` and addition owner fields if needed.

Closing https://github.com/elastic/package-registry/issues/451

### CODE REVIEW

**Comment by ruflin** [comment]
@andresrc @mtojek If we move forward with this, I suggest we update the packages to have the owner team on Github inside this field.

**Comment by mtojek** [comment]
To be honest, I'm against this change to prevent community people from pinging single developers instead of using team aliases. There is no plan for fixing this field if the person leaves the company and we won't fix already released packages.

I suggest to base on the repository owners here.

**Comment by ruflin** [comment]
I used `owner.github` to exactly allow teams in here. For example `owner.github: elastic/integrations-platforms`.

File: dev/packages/example/reference/1.0.0/manifest.yml [hunk 1]
```diff
@@ -28,6 +28,10 @@ license: basic
 # The default type is integration and will be set if empty.
 type: integration
 
+# Details about the owner of the package. Current only github link is supported.
+# A package can only have a single owner, the owner can be a user or a team.
```
Comment (ruflin) [inline dev/packages/example/reference/1.0.0/manifest.yml:32]: @mtojek I added a comment here to make it more obvious it can be a user or a team.

Comment (mtojek) [reply dev/packages/example/reference/1.0.0/manifest.yml:32]: Thanks for the explanation. I assume that team aliases will be set for all Elastic packages?

Comment (ruflin) [reply dev/packages/example/reference/1.0.0/manifest.yml:32]: Yes, I hope so. This was the initial driver for this in https://github.com/elastic/package-registry/issues/451 Not sure if there is a good way to add this to the generator as in Filebeat for example, some belong to SIEM, some to Platforms team.

**Comment by ruflin** [comment]
After going through a few merge conflicts and rebases, this should now be ready again @mtojek

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Support multiple packages paths. [#525](https://github.com/elastic/package-registry/pull/525)
 * Added support for ecs style validation for dataset fields. [#520](https://github.com/elastic/package-registry/pull/520)
 * Use BasePackage for search output data. [#529](https://github.com/elastic/package-registry/pull/529)
+* Add support for owner filed in package manifest. [#535](https://github.com/elastic/package-registry/pull/535)
```
Comment (mtojek) [inline CHANGELOG.md:35]: nit: field

Comment (mtojek) [reply CHANGELOG.md:35]: nit: 536

Comment (ruflin) [reply CHANGELOG.md]: fixed. that is what happens if you try to predict the PR number ...

**Pull request merged**